addons/account/models/account_bank_statement.py

```diff
@@ -556,9 +556,14 @@ def process_reconciliation(self, counterpart_aml_dicts=None, payment_aml_rec=Non
             raise UserError(_('A selected statement line was already reconciled with an account move.'))
 
         # Fully reconciled moves are just linked to the bank statement
+
         total = self.amount
+        company = self.company_id
+        date = self.date or fields.Date.today()
+        currency = self.currency_id or statement_currency
+        company_total = currency._convert(total, company_currency, company, date)
         for aml_rec in payment_aml_rec:
-            total -= aml_rec.debit - aml_rec.credit
+            company_total -= aml_rec.debit - aml_rec.credit
```

Comment:
Something like that is more correct I think:
```
balance = aml_rec.currency_id and aml_rec.amount_currency or aml_rec.balance
total -= balance
```
Can you also write a test for this plz?

@qdp-odoo Using the `amount_currency` supposes the payment and the statement line are always in the same currency. Is it always the case?

Reply:
@smetl 

You can use the amount_currency if the bank statement and payment are using the same currency. But what about other cases?
e.g.
Bank Statement: USD
Payment: EUR
Company: MXN.

If we use amount_currency we must show a message(Something like "You cannot complete this process because...") when someone tries to reconcile payments and bank statements with different currencies.

I think we have two possibles solutions here. Adding a solution that solves as many cases as possible or stopping the reconciliation process when these cases happen(With its detailed error message).

Regards.

Reply:
@smetl any feedback?

Reply:
@josemoralesp So, in my opinion, the code should be something like:

```
currency = ....
for aml_rec in payment_aml_rec:
  balance = aml_rec.currency_id and aml_rec.amount_currency or aml_rec.balance
  aml_currency = aml_rec.currency_id or aml_rec.company_currency_id
  total -= aml_currency._convert(balance, currency, aml_rec.company_id, aml_rec.date)
```

Reply:
@smetl I've added the change suggested by you.